### ISSUE TITLE
Specific shortcut key not working

### ISSUE THREAD

**Issue by crazydef** [microsoft/vscode#176086]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version:

Version: 1.75.1 (Universal)
Commit: 441438abd1ac652551dbe4d408dfcec8a499b8bf
Date: 2023-02-08T21:34:59.000Z
Electron: 19.1.9
Chromium: 102.0.5005.194
Node.js: 16.14.2
V8: 10.2.154.23-electron.0
OS: Darwin arm64 22.1.0
Sandboxed: No

- OS Version: 
MacOS Ventura Version 13.0 (22A8381)

Steps to Reproduce:

So my steps to reproduce are basically use VSCode, but my circumstances are a bit unusual, and it's highly unlikely anyone will be able to reproduce this, but it's an issue that's highly frustrating.  So I'm hoping it's something obvious that I've missed...

Anyway...  I'm using VSCode on a new Mac, connecting remotely with NoMachine.  Everything works fine, except for this one keyboard shortcut:  Ctrl+Shift+Right

I have it configured in the keyboard settings as follows:
```
    {
        "key": "ctrl+shift+right",
        "command": "cursorWordStartRightSelect",
        "when": "textInputFocus"
    },
```

The only other references to that shortcut combination in that file are:
```
    {
        "key": "ctrl+shift+right",
        "command": "-cursorWordEndRightSelect",
        "when": "textInputFocus"
    },
    {
        "key": "ctrl+shift+right",
        "command": "-editor.action.smartSelect.expand",
        "when": "editorTextFocus"
    },
```

I have three similar entries for Ctrl+Shift+Left, and that works fine.  Ctrl+Right works, Shift+Right works, but not Ctrl+Shift+Right.  I know it's not a keyboard/NoMachine issue, and I've disabled all known OS shortcuts that might be taking precedence.  In fact Ctrl+Shift+Right works just fine in other Mac applications.  (Like TextEdit, and XCode, for example.)  Just not in VSCode.

I've tried switching to Keyboard Dispatch to KeyCode.  I tried switching off Editor: Accessibility Support.  Nothing works.

And so I turn to you good folks, hoping someone, somewhere has experienced something similar and knows what the problem is...

**Comment by IllusionMH** [comment]
Have you tried steps from https://github.com/microsoft/vscode/wiki/Keybinding-Issues to see if VS Code receives this shortcut and what it attempts to do.

**Comment by crazydef** [comment]
So with `Keyboard Shortcuts Troubleshooting` enabled I see no event for the shortcut in question.  I get events for the shift and control keys being pressed, but nothing for the right arrow key afterwards.

Digging a bit further into this, I connected a physical keyboard to my Mac and installed Key Codes.  Key Codes detects all events for all keys, but the modifier keys sent from the physical keyboard are different to the key inputs sent through NoMachine.

The physical keyboard sends:

Unicode: 63235 / 0xF703
Key Code: 124 / 0x7C
Modifiers: 10879235 / 0xA60103

while NoMachine sends:

Unicode: 63235 / 0xF703
Key Code: 124 / 0x7C
Modifiers: 8782083 / 0x860103

But that said, I have an older iMac here (with an older macOS) that receives the same key inputs through both NoMachine and the physical keyboard and VSCode works fine on that machine.  Which presumably indicates it's somebody else's problem as far as you're concerned - which I kinda suspected anyway, but it's a bit strange that other applications still work with that combination.

**Comment by crazydef** [comment]
In a rather bizarre turn of events, Ctrl+Shift+Left has now also stopped working.  Also only in VSCode.

(I noticed it when I turned my Mac on this morning.  A few minutes later, VSCode reported it was corrupted and told me to reinstall.  I installed the latest version through the `Check for Updates` menu, and while VSCode doesn't think it has been corrupted now, both shortcuts remain unavailable.)

Are there hidden/undocumented features in Chrome that could be affecting this?
